spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/AbstractMqttMessageHandler.java

```diff
@@ -134,7 +134,7 @@ public boolean isRunning() {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.connectIfNeeded();
+		connectIfNeeded();
```

Comment:
Bug? Not sure, because method is `@Deprecated`, so any its usage is crossed out. 
Right, I agree that even the Gradle I don't see a WARN. 
Maybe just because both these methods are `protected` and only in this class and compiler just doesn't see vulnerabilities...
So, JDK bug! :smile: